bugsnag-android-core/src/androidTest/java/com/bugsnag/android/StacktraceTest.java

```diff
@@ -1,118 +0,0 @@
-        assertFalse(firstFrame.has("inProject"));
-    }
-
-    @Test
-    public void testInProject() throws JSONException, IOException {
-        config.setProjectPackages(Collections.singleton("com.bugsnag.android"));
-        Collection<String> projectPackages = config.getProjectPackages();
-        Stacktrace stacktrace = new Stacktrace(exception.getStackTrace(), projectPackages);
-        JSONArray stacktraceJson = streamableToJsonArray(stacktrace);
-
-        JSONObject firstFrame = (JSONObject) stacktraceJson.get(0);
-        assertTrue(firstFrame.getBoolean("inProject"));
-    }
-
-    @Test
-    public void testStacktraceTrimming() throws Throwable {
-        List<StackTraceElement> elements = new ArrayList<>();
-
-        for (int k = 0; k < 1000; k++) {
-            elements.add(new StackTraceElement("SomeClass",
-                "someMethod", "someFile", k));
-        }
-
-        StackTraceElement[] ary = new StackTraceElement[elements.size()];
-        Stacktrace stacktrace = new Stacktrace(elements.toArray(ary),
-                Collections.<String>emptyList());
-        JSONArray jsonArray = streamableToJsonArray(stacktrace);
-        assertEquals(200, jsonArray.length());
-        assertEquals(0, jsonArray.getJSONObject(0).getInt("lineNumber"));
-        assertEquals(199, jsonArray.getJSONObject(199).getInt("lineNumber"));
-    }
-
-    @Test
-    public void testStacktraceTrimmingListCtor() throws Throwable {
-        List<Map<String, Object>> elements = new ArrayList<>();
-
-        for (int k = 0; k < 1000; k++) {
-            elements.add(Collections.singletonMap("Foo", (Object) k));
-        }
-
-        Stacktrace stacktrace = new Stacktrace(elements);
-        JSONArray jsonArray = streamableToJsonArray(stacktrace);
-        assertEquals(200, jsonArray.length());
-        assertEquals(0, jsonArray.getJSONObject(0).getInt("Foo"));
-        assertEquals(199, jsonArray.getJSONObject(199).getInt("Foo"));
-    }
-
-    @Test
-    public void testClassNameResolution() throws JSONException, IOException {
-        StackTraceElement[] stackTraceElements = {
-            new StackTraceElement("SomeClass", "someMethod",
-                    "someFile", 12)};
-        Stacktrace stacktrace = new Stacktrace(stackTraceElements,
-                Collections.<String>emptyList());
-        JSONArray stacktraceJson = streamableToJsonArray(stacktrace);
-
-        JSONObject frame = (JSONObject) stacktraceJson.get(0);
-        assertEquals("SomeClass.someMethod", frame.get("method"));
-
-        StackTraceElement stackTraceElement = new StackTraceElement("",
-            "someMethod", "someFile", 12);
-        Stacktrace stacktrace1 = new Stacktrace(
-            new StackTraceElement[]{stackTraceElement}, Collections.<String>emptyList());
-        stacktraceJson = streamableToJsonArray(stacktrace1);
-
-        frame = (JSONObject) stacktraceJson.get(0);
-        assertEquals("someMethod", frame.get("method"));
-    }
-}
```

Comment:
This test appears to be missing in the refactor.

Reply:
This particular test case is now implicit, as there is now a JSON fixture which holds the expected output. If the class name/method name/file name is wrong, the generated JSON will be wrong and the test will fail as the two are different.